pallets/msa/src/lib.rs

```diff
@@ -862,7 +860,7 @@ pub mod pallet {
 					);
 				},
 			}
-			Ok(Some(T::WeightInfo::retire_msa(num_deletions)).into())
+			Ok(Some(T::WeightInfo::retire_msa()).into())
```

Comment:
As the weight is static, I think we can now just return `Ok`. (and update `DispatchResultWithPostInfo` to just `DispatchResult`)
```suggestion
			Ok()
```

Reply:
I think this needs to be `Ok(())`